### PR TITLE
further optimizations

### PR DESCRIPTION
**Pull request by silkentrance**
I have some more optimizations ready that brings this, given the available micro benchmark, to around 30ms instead of approx 40ms without these modifications. I consider the remaining 2-3ms overhead to be related to the generator/yield feature when comparing it to linebyline which achieves approx. 28ms.

The optimizations center around increasing the default buffer size to 64KiB and getting rid of atend and seenCR and instead replacing these by boolean expressions that the compiler is able to optimize better than when pre-calculating these values on every run of the inner loop.
In addition I moved the seenCR to an else if branch that will only be executed once for every readChunk and not every time the inner loop reiterates.

And the good thing is, the test cases won't fail.

But please, before publishing this to npm, make sure that everything is the way you want it to be, ok?

### CODE REVIEW

**Comment by neurosnap** [comment]
Thanks for your help, I'll take a deeper look tonight.

**Comment by silkentrance** [comment]
Take your time.

File: index.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ var CR = 13;
  * @return {Object} The generator object
  */
 function* readlines(fd, filesize, bufferSize, position) {
-  if (typeof bufferSize === 'undefined') bufferSize = 1024;
+  if (typeof bufferSize === 'undefined') bufferSize = 64 * 1024;
```
Comment (neurosnap) [inline index.js:18]: Any reason in particular why you changed the default buffer size?

Comment (silkentrance) [inline index.js:18]: The reason being that this is much faster with larger files as we do not hit the disk that often. In addition, default page size is 4KiB when considering memory mapped disk I/O.

For smaller files, the guard you installed that reduces the buffer size to the remaining size will limit the read chunk buffer size to the actual file size.

Furthermore, for fs.createReadStream() the default buffer size is also 64KiB.

So, for files smaller than 64KiB, the whole file content will be read in one single read operation, increasing performance by approx. 4-5 ms without any of the other optimizations in place.

**Pull request merged**